src/display-buffer.coffee

```diff
@@ -841,9 +841,15 @@ class DisplayBuffer extends Model
     if screenLine.isSoftWrapped() and column >= maxScreenColumn
       if wrapAtSoftNewlines
         row++
-        column = 0
+        column = @screenLines[row].clipScreenColumn(0)
       else
         column = screenLine.clipScreenColumn(maxScreenColumn - 1)
+    else if screenLine.isColumnInsidePhantomToken(column)
+      if wrapAtSoftNewlines
```

Comment:
Sure enough, when I invert this logic and only move forward when `wrapAtSoftNewlines` is `true`, I no longer need to pass an option in `moveLeft`. However, it breaks some existing specs that I *think* might actually be wrong. The are all specs for clipping the position at the beginning of a continuation of a soft-wrapped line... Pretty sure we should always back up by 1 column when wrapping back up before the soft-wrap so we're not leaving the cursor on an invalid position.